Library/Homebrew/software_spec.rb

```diff
@@ -402,20 +403,28 @@ def compatible_locations?
   # Does the {Bottle} this {BottleSpecification} belongs to need to be relocated?
   sig { returns(T::Boolean) }
   def skip_relocation?
-    cellar == :any_skip_relocation
+    bottle_checksum = checksum_for(Utils::Bottles.tag).first
```

Comment:
The `.first`s in a few places feel weird; why is an array being used here if we're discarding the other values?

Reply:
I changed the code to try to find the sha256 by it's length.

The issue is that the hash has two keys: the sha256, and "cellar" (optional). This is why I used first and second here, but this is gone now with my new strategy.